packages/@lwc/engine-core/src/renderer.ts

```diff
@@ -0,0 +1,283 @@
+    querySelectorAll = querySelectorAllImpl;
+}
+
+type getElementsByTagNameFunc = (element: E, tagNameOrWildCard: string) => HTMLCollection;
+export let getElementsByTagName: getElementsByTagNameFunc;
+export function setGetElementsByTagName(getElementsByTagNameImpl: getElementsByTagNameFunc) {
+    getElementsByTagName = getElementsByTagNameImpl;
+}
+
+type getElementsByClassNameFunc = (element: E, names: string) => HTMLCollection;
+export let getElementsByClassName: getElementsByClassNameFunc;
+export function setGetElementsByClassName(getElementsByClassNameImpl: getElementsByClassNameFunc) {
+    getElementsByClassName = getElementsByClassNameImpl;
+}
+
+type getChildrenFunc = (element: E) => HTMLCollection;
+export let getChildren: getChildrenFunc;
+export function setGetChildren(getChildrenImpl: getChildrenFunc) {
+    getChildren = getChildrenImpl;
+}
+
+type getChildNodesFunc = (element: E) => NodeList;
+export let getChildNodes: getChildNodesFunc;
+export function setGetChildNodes(getChildNodesImpl: getChildNodesFunc) {
+    getChildNodes = getChildNodesImpl;
+}
+
+type getFirstChildFunc = (element: E) => N | null;
+export let getFirstChild: getFirstChildFunc;
+export function setGetFirstChild(getFirstChildImpl: getFirstChildFunc) {
+    getFirstChild = getFirstChildImpl;
+}
+
+type getFirstElementChildFunc = (element: E) => E | null;
+export let getFirstElementChild: getFirstElementChildFunc;
+export function setGetFirstElementChild(getFirstElementChildImpl: getFirstElementChildFunc) {
+    getFirstElementChild = getFirstElementChildImpl;
+}
+
+type getLastChildFunc = (element: E) => N | null;
+export let getLastChild: getLastChildFunc;
+export function setGetLastChild(getLastChildImpl: getLastChildFunc) {
+    getLastChild = getLastChildImpl;
+}
+
+type getLastElementChildFunc = (element: E) => E | null;
+export let getLastElementChild: getLastElementChildFunc;
+export function setGetLastElementChild(getLastElementChildImpl: getLastElementChildFunc) {
+    getLastElementChild = getLastElementChildImpl;
+}
+
+type isConnectedFunc = (node: N) => boolean;
+export let isConnected: isConnectedFunc;
+export function setIsConnected(isConnectedImpl: isConnectedFunc) {
+    isConnected = isConnectedImpl;
+}
+
+type insertGlobalStylesheetFunc = (content: string) => void;
+export let insertGlobalStylesheet: insertGlobalStylesheetFunc;
+export function setInsertGlobalStylesheet(insertGlobalStylesheetImpl: insertGlobalStylesheetFunc) {
+    insertGlobalStylesheet = insertGlobalStylesheetImpl;
+}
+
+type insertStylesheetFunc = (content: string, target: N) => void;
+export let insertStylesheet: insertStylesheetFunc;
+export function setInsertStylesheet(insertStylesheetImpl: insertStylesheetFunc) {
+    insertStylesheet = insertStylesheetImpl;
+}
+
+type assertInstanceOfHTMLElementFunc = (elm: any, msg: string) => void;
+export let assertInstanceOfHTMLElement: assertInstanceOfHTMLElementFunc;
```

Comment:
It seem that this trick is enough to overcome the issue with TS, exporting something that is never set during initialization, but has a specific type makes it more difficult to TS to assume the problem of being undefined when used from another file, nice!